src/components/Nfts/NftCard.tsx

```diff
@@ -18,49 +19,48 @@ import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingl
 import { useModal } from 'hooks/useModal/useModal'
 import { getMixPanel } from 'lib/mixpanel/mixPanelSingleton'
 import { MixPanelEvents } from 'lib/mixpanel/types'
-import type { SupportedZapperNetwork, V2ZapperNft } from 'state/apis/zapper/validators'
-import { getMediaType, zapperNetworkToChainId } from 'state/apis/zapper/validators'
+import type { NftItem } from 'state/apis/nft/types'
+import { getMediaType } from 'state/apis/zapper/validators'
 import { selectAssetById } from 'state/slices/selectors'
 import { useAppSelector } from 'state/store'
 
 type NftCardProps = {
-  zapperNft: V2ZapperNft
+  nftItem: NftItem
 }
 
-export const NftCard: React.FC<NftCardProps> = ({ zapperNft }) => {
-  const { collection, medias, name, rarityRank } = zapperNft
-  const { floorPriceEth } = collection
+export const NftCard: React.FC<NftCardProps> = ({ nftItem }) => {
+  const { collection, medias, name, rarityRank } = nftItem
+  const { floorPrice } = collection
   const mediaUrl = medias?.[0]?.originalUrl
   const mediaType = getMediaType(mediaUrl)
   const bg = useColorModeValue('gray.50', 'gray.750')
   const bgHover = useColorModeValue('gray.100', 'gray.700')
   const placeholderImage = useColorModeValue(PlaceholderDrk, Placeholder)
   const [isMediaLoaded, setIsMediaLoaded] = useState(false)
 
-  const maybeNetwork = collection?.network
-  const maybeChainId = zapperNetworkToChainId(maybeNetwork as SupportedZapperNetwork)
-  const maybeChainAdapter = getChainAdapterManager().get(maybeChainId as ChainId)
+  const chainId = collection.chainId
+  const maybeChainAdapter = getChainAdapterManager().get(chainId as ChainId)
   const maybeFeeAssetId = maybeChainAdapter?.getFeeAssetId()
   const maybeFeeAsset = useAppSelector(state => selectAssetById(state, maybeFeeAssetId ?? ''))
 
-  const { nft } = useModal()
+  const { nft: nftModal } = useModal()
 
   const handleClick = useCallback(() => {
-    nft.open({ zapperNft })
+    nftModal.open({ nftItem })
 
     const mixpanel = getMixPanel()
     const eventData = {
-      name: zapperNft.name,
-      id: zapperNft.id,
-      collectionName: zapperNft.collection.name,
-      collectionAddress: zapperNft.collection.address,
-      estimatedValueEth: zapperNft.estimatedValueEth,
-      collectionFloorPriceEth: zapperNft.collection.floorPriceEth,
-      nftMediaUrls: (zapperNft.medias ?? []).map(media => media.originalUrl),
+      name: nftItem.name,
+      id: nftItem.id,
+      collectionName: nftItem.collection.name,
+      collectionAddress: fromAssetId(nftItem.collection.id!).assetReference,
```

Comment:
are we sure there will be a collection id? would it be better to use a ternary to fallback to `''` because `fromAssetId` will throw if empty string I'm pretty sure

Reply:
As discussed, this is a bit weird but yeah, depending on the endpoint being hit, we may or may not have a collection ID - eventually when we reconciliate the data from diff. endpoints, we will have an `id`.
A collection ID (a collection contract's `AssetId`) is the bare minimum data for a collection, however, some endpoints like that Zapper one just return us a weirdly denormalized response with no `id`.

As we consolidate endpoints calls and parsing, we will be able to remove a lot of the optional overhead